ctapipe/reco/shower_reconstructor.py

```diff
@@ -39,6 +37,10 @@ def __call__(self, event):
         self.tel_positions = tel_positions
         self.tel_solid_angles = tel_solid_angles
         self.tel_mirror_area = tel_mirror_area
+        self.tel_pix_coords_altaz = self._tel_pix_coords_altaz(event)
+
+        # for likelihood
+        self.DL1CamContainers = event.dl1.tel.items()
```

Comment:
should not use PascalCase here, since it is a variable so pep8 says to use snake_case. (so `self.dl1_camera_containers`

Reply:
Do we really need to store that as state?

Reply:
Good point. There are two choices I think. Since `likelihood` is only called by Minuit, which only allows `(self, params)` as arguments of the likelihood, it has to either know the dl1 cam containers or the event structure. So its either `self.d1_cam_containers = event.dl1.tel.items()` or `self.event = event`. Which one should I use?

Reply:
Third option would be to just store the dl1 images for the likelihood and the whole container

Reply:
This is a classic case for using a closure, e.g. via partial:


```
def likelihood(..., images):
   ...

m  = Minuit(partial(likelihood, images=mages), ...)
```

Reply:
Ah great, I didn't know this. I'll change it